src/inmanta/deploy/scheduler.py

```diff
@@ -673,15 +678,22 @@ async def report_resource_state(
                     state.deployment_result = deployment_result
                 return
 
-            if skipped_for_dependency:
-                state.blocked = BlockedStatus.TRANSIENT
             # We are not stale
             state.status = status
             if deployment_result is not None:
                 # first update state, then send out events
                 self._deploying_latest.remove(resource)
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
+                # Check if we need to mark a resource as transiently blocked
+                # Or if we can unblock a transiently blocked resource (its dependencies now succeed)
+                if skipped_for_dependency and state.blocked is not BlockedStatus.YES:
```

Comment:
I was writing a comment about how I thought you were mixing up two scenarios here, until I realized that you're covering [this case](https://github.com/inmanta/inmanta-core/pull/8383#discussion_r1862339660) here. I think the implementation is good (with the same caveat as in `_new_version()` that I think we should check compliance instead), but I think it definitely needs a comment as to *why* we're doing this. i.e. if the task reports `skipped_for_dependency`, why don't we simply believe it?